Source/WebCore/loader/FrameLoader.h

```diff
@@ -457,7 +457,7 @@ class FrameLoader final : public CanMakeCheckedPtr<FrameLoader> {
     void updateNavigationAPIEntries();
     void updateRequestAndAddExtraFields(Frame&, ResourceRequest&, IsMainResource, FrameLoadType, ShouldUpdateAppInitiatedValue, IsServiceWorkerNavigationLoad, WillOpenInNewWindow, Document*);
 
-    bool dispatchNavigateEvent(const URL& newURL, FrameLoadType, const NavigationAction&, NavigationHistoryBehavior, bool isSameDocument);
+    bool dispatchNavigateEvent(const URL& newURL, FrameLoadType, const NavigationAction&, NavigationHistoryBehavior, bool isSameDocument, FormState* = nullptr);
```

Comment:
We should pass a `RefPtr<FormState>` instead of a raw pointer.

Reply:
I believe this fits the style other reviewers have recommended.

Either pass as a `RefPtr&&` if transferring ownership or a `*` otherwise. This is not transferring ownership.

Reply:
Ah you're right. I was thinking of this rule: https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines#when-passing-an-object-to-non-trivial-function-hold-a-smart-pointer-to-this-object-on-the-stack


But in this case I think it is ok as you have written because all the call sites have a RefPtr on the stack as the function argument.

Reply:
What actually happened was in my head I combined the linked rule and the previous one dealing with calling functions. Anyway, yeah this is fine as written.